### PR TITLE
feature/eslint

### PR DESCRIPTION
**Pull request by Le-Caignec**
Sohint linter for Smart contract raised some warning about the way that we import our smart contract : [this is the corresponding rule](https://protofire.github.io/solhint/docs/rules/best-practises/no-global-import.html).
Some solutions :

keep code like that
disable the rule
change the import to : import {A} from "./A.sol" => that will create a difference with what is currently in production

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
+    "plugin:@typescript-eslint/recommended",
+    "prettier"
+  ],
+  "rules": {
+    "import/no-extraneous-dependencies": [
+      "error",
+      {
+        "devDependencies": true
+      }
+    ]
```
Comment (PierreJeanjacquot) [inline .eslintrc.json]: `import/no-extraneous-dependencies` accepting `devDependencies` should apply only to the `scripts`and `test` files

**Review by PierreJeanjacquot — APPROVED** [review]
It may be worth to add lint step to the pipeline now.

In the future, we may want to apply `import/no-extraneous-dependencies` including `devDependencies` only to relevant files.
Also, fixing the solidity import pattern in future PR seems OK.

File: .eslintrc.json (hunk 1 above)
Comment (Le-Caignec) [reply .eslintrc.json]: I can fix it now but ` @nomicfoundation/hardhat-toolbox`  & `hardhat` should be set as `dependency` and not `dev-dependency`

Comment (Le-Caignec) [reply .eslintrc.json]: Commit: 7f264c6b53878c50d596cf2684b64d4c2a102e88
Fix

**Pull request merged**